extension/src/util/string.ts

```diff
@@ -1 +1,8 @@
-export const shortenForLabel = (str: string): string => str.slice(0, 7)
```

Comment:
[F] This is where the error was happening, you cannot slice `null`, simply adding an optional chain seemed a little dicey so I have updated the incoming types and flushed out some issues in `extension/src/experiments/columns/extract.ts`. The resulting function is a complete monstrosity but backwards compatible so I can live with it. Please LMK if you have better ideas... especially for the file name.